types/wlr_surface.c

```diff
@@ -67,6 +67,9 @@ static void surface_set_opaque_region(struct wl_client *client,
 		struct wl_resource *resource,
 		struct wl_resource *region_resource) {
 	struct wlr_surface *surface = wl_resource_get_user_data(resource);
+	if ((surface->pending.invalid & WLR_SURFACE_INVALID_OPAQUE_REGION)) {
+		pixman_region32_clear(&surface->pending.opaque);
```

Comment:
I think it's fine to copy.
wayland-server-protocol.h says this for damage, which is clearly union:
```
         * The initial value for pending damage is empty: no damage.
         * wl_surface.damage adds pending damage: the new pending damage is
         * the union of old pending damage and the given rectangle.
```

On the other hand, opaque and input have nothing about accumulating, instead they just say 'copy'
For opaque
```
         * The initial value for an opaque region is empty. Setting the
         * pending opaque region has copy semantics [...]
```
And input
```
         * The initial value for an input region is infinite. That means
         * the whole surface will accept input. Setting the pending input
         * region has copy semantics...
```